DigitalLearningSolutions.Web/Controllers/RegisterController.cs

```diff
@@ -0,0 +1,169 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers
+{
+    using System;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Extensions;
+    using DigitalLearningSolutions.Web.Models;
+    using DigitalLearningSolutions.Web.ServiceFilter;
+    using DigitalLearningSolutions.Web.ViewModels.Register;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+
+    public class RegisterController : Controller
+    {
+        private const string CookieName = "RegistrationData";
+        private readonly ICentresService centresService;
+        private readonly IJobGroupsService jobGroupsService;
+
+        public RegisterController(ICentresService centresService, IJobGroupsService jobGroupsService)
+        {
+            this.centresService = centresService;
+            this.jobGroupsService = jobGroupsService;
+        }
+
+        public IActionResult Index()
+        {
+            if (User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home");
+            }
+
+            var delegateRegistrationData = TempData.Get<DelegateRegistrationData>();
+
+            if (delegateRegistrationData == null || !Request.Cookies.ContainsKey(CookieName))
+            {
+                var id = Guid.NewGuid();
+                delegateRegistrationData = new DelegateRegistrationData { Id = id };
+
+                Response.Cookies.Append(
+                    CookieName,
+                    id.ToString(),
+                    new CookieOptions
+                    {
+                        Expires = DateTimeOffset.UtcNow.AddDays(30)
+                    });
+                TempData.Set(delegateRegistrationData);
+
+                return View();
+            }
+
+            return View(delegateRegistrationData.RegisterViewModel);
+        }
+
+        [HttpPost]
+        public IActionResult Index(RegisterViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
+
+            var data = TempData.Get<DelegateRegistrationData>();
+            data.RegisterViewModel = model;
+            TempData.Set(data);
+
+            return RedirectToAction("LearnerInformation");
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<DelegateRegistrationData>))]
+        [Route("Register/learner-information")]

```

Comment:
Is there a reason we're overriding the default routes here?

Reply:
Good question 🤔
I guess the answer's no, I'll take them out